### PR TITLE
[ENHANCEMENT]  Implement Skip prompt flag on datasource profile cli

### PR DESCRIPTION
**Pull request by thcidale0808**
The aim of this PR is to create an option to skip user prompt in the datasource profile cli where the user has only one data source.

Changes proposed in this pull request:
- Added the -y/--skip_prompt option to click command decorator
- Created a generic function for the conditional logic to determine whether to prompt or not

Addresses [#1690](https://github.com/great-expectations/great_expectations/issues/1690).

### CODE REVIEW

**Comment by thcidale0808** [comment]
@cla-bot check

File: great_expectations/cli/datasource.py [hunk 1]
```diff
@@ -181,6 +181,13 @@ def _build_datasource_intro_string(datasource_count):
     help="Profile ALL data assets within the target data source. "
     "If True, this will override --max_data_assets.",
 )
+@click.option(
+    "--skip_prompt",
+    "-y",
```
Comment (jcampbell) [inline great_expectations/cli/datasource.py]:
```suggestion
    "--assume-yes/--yes",
    "-y",
```

Small requested word change here, which will have a cascading (small) change throughout the PR

**Review by jcampbell — CHANGES_REQUESTED** [review]
Thank you! I requested a small wording change.

File: great_expectations/cli/datasource.py (hunk 1 above)
Comment (thcidale0808) [reply great_expectations/cli/datasource.py]: @jcampbell it's fixed!

**Comment by jcampbell** [comment]
Thank you!

**Review by jcampbell — APPROVED** [review]

**Pull request merged**